src/AzureRenderManager/WebApp/Arm/AzureResourceProvider.cs

```diff
@@ -70,14 +72,39 @@ public AzureResourceProvider(IHttpContextAccessor contextAccessor) : base(contex
             var token = new TokenCredentials(accessToken);
             var authClient = new AuthorizationManagementClient(token) { SubscriptionId = subscriptionId.ToString() };
 
+            var isClassicAdminTask = IsCurrentUserClassicAdministrator(authClient);
             var result = await GetRoleDefinitions(authClient, $"/subscriptions/{subscriptionId}");
             var roleDefs = result.Where(rd =>
                 rd.Permissions.Any(p =>
                     p.Actions.Any(a => ActionsThatCanCreate.Contains(a.ToLower(CultureInfo.InvariantCulture))))).ToList();
 
             var subscriptionRoles = await GetRoleAssignmentsForCurrentUser(authClient, subscriptionScope, roleDefs);
+            var isClassicAdmin = await isClassicAdminTask;
 
-            return subscriptionRoles.Any(ra => roleDefs.Any(rd => rd.Id == ra.RoleDefinitionId));
+            return isClassicAdmin || subscriptionRoles.Any(ra => roleDefs.Any(rd => rd.Id == ra.RoleDefinitionId));
+        }
+
+        private async Task<bool> IsCurrentUserClassicAdministrator(AuthorizationManagementClient authClient)
+        {
+            var result = await authClient.ClassicAdministrators.ListAsync();
+            var classicAdmins = result.ToList();
+            var user = GetUser();
+            var names = user.Identities.Select(i => i.Claims.GetName());
+            var emails = user.Identities.Select(i => i.Claims.GetEmailAddress());
+            var upns = user.Identities.Select(i => i.Claims.GetUpn());
+            foreach (var adminEmail in GetClassicAdministratorEmails(classicAdmins))
+            {
+                if (names.Any(name => name.Contains(adminEmail)) || upns.Contains(adminEmail) || emails.Contains(adminEmail))
```

Comment:
isn't `name.Contains(adminEmail)` insecure? e.g. admin is `admin@example.com` and I register `notanadmin@example.com`?

Reply:
is this meant to be `names.Contains(adminEmail)` like the others?

Reply:
It's not insecure in the sense that this is only use for a warning dialog, all calls are still done using delegated permissions and would fail if the user didn't have the required permissions.  The issue was around the prefix which gets tacked on so my method was not very reliable :-)

Reply:
Ok, so this doesn't grant access to anything...